support/reftest/reftest.py

```diff
@@ -0,0 +1,310 @@
+        default=None,
+        help="Access ID for Amazon services. If no ID is provided, attempts to"
+        " find it among environment variables and ~/.aws/config file will"
+        " be made",
+    )
+    parser.add_argument(
+        "--aws-access-key",
+        default=None,
+        help="Access key for Amazon services. If no key is provided, attempts"
+        " to find it among environment variables and ~/.aws/config file"
+        " will be made",
+    )
+    parser.add_argument(
+        "--aws-session-token",
+        default=None,
+        help="Session token for Amazon services. If no token is provided,"
+        " attempts to find it among environment variables and"
+        " ~/.aws/config file will be made",
+    )
+    parser.add_argument(
+        "--default-region",
+        default=None,
+        help="Default region for Amamzon services. If no region is provided,"
+        " attempts to find it among environment variables and"
+        " ~/.aws/config file will be made",
+    )
+
+
+def parse_cdn_certs(parser):
+    parser.add_argument(
+        "--cert", type=str, help="Client certificate file and password", required=True
+    )
+    parser.add_argument("--key", type=str, help="Private key file name", required=True)
+    parser.add_argument(
+        "--cacert",
+        type=str,
+        help="CA certificate to verify peer against",
+        required=True,
+    )
+
+
+def parse_args():
+    root_parser = argparse.ArgumentParser(description=__doc__)
+    subparsers = root_parser.add_subparsers(
+        dest="command", help="reference test data operations"
+    )
+    parser_prepare = subparsers.add_parser(
+        "prepare",
+        help="""
+        prepare referece test data for intagration test
+         e.g. $./reftest.py prepare --cert rcm-debug.crt --key rcm-debug.key --cacert redhat-uep.pem
+        """,
+    )
+    parser_prepare.add_argument(
+        "--release-date",
+        default=None,
+        help="Date on which the content will be made available.",
+    )
+    parse_aws_info(parser_prepare)
+    parse_cdn_certs(parser_prepare)
+
+    parser_clean = subparsers.add_parser(
+        "clean",
+        help="""
+        clean referece test data in s3 and dynamodb
+        e.g. $./reftest.py clean
+        """,
+    )
+    parse_aws_info(parser_clean)
+
+    return root_parser.parse_args()
+
+
+class RefTestConfig:
+    def __init__(self, prod_cdn_url, content_paths, table, bucket):
+        self.table = table
+        self.bucket = bucket
+        self.prod_cdn_url = prod_cdn_url
+        self.content_paths = content_paths
+
+
+def load_config():
+    with open("data.yml") as f:
+        config = yaml.load(f, yaml.SafeLoader)
+    return RefTestConfig(
+        config["prod-cdn-url"],
+        config["path"],
+        config["destination"]["table"],
+        config["destination"]["bucket"],
+    )
+
+
+def download_to_local(url, key_path, cert_path, cacert_path):
+    local_filename = url.split("/")[-1]
```

Comment:
This seems rather dangerous.  You're downloading into the current working directory and overwriting whatever's already there with the same filename.  So, for instance, if I have a file named "listing" in the current directory when I run this command, it's going to be silently deleted.

Should this use a temporary file instead?  (e.g. NamedTemporaryFile)

Reply:
> This seems rather dangerous. You're downloading into the current working directory and overwriting whatever's already there with the same filename. So, for instance, if I have a file named "listing" in the current directory when I run this command, it's going to be silently deleted.
> 
> Should this use a temporary file instead? (e.g. NamedTemporaryFile)

@rohanpm solved.